avocado/core/runner.py

```diff
@@ -506,10 +507,19 @@ def _template_to_factory(template, variant):
         params = variant.get("variant"), variant.get("mux_path")
         if params:
             if "params" in template[1]:
-                msg = ("Unable to use test variants %s, params are already"
-                       " present in test factory: %s"
-                       % (template[0], template[1]))
-                raise ValueError(msg)
+                if not varianter.is_empty_variant(params[0]):
+                    msg = ("Specifying test params from test loader and "
+                           "from varianter at the same time is not yet "
+                           "supported. Please remove either variants defined"
+                           "by the varianter (%s) or make the test loader of"
+                           "test %s to not to fill variants."
+                           % (variant, template))
+                    raise NotImplementedError(msg)
+                params = template[1]["params"]
+                variant_id = varianter.generate_variant_id(params[0])
+                return template, {"variant": params[0],
+                                  "variant_id": variant_id,
+                                  "mux_path": params[1]}
```

Comment:
This was the target implementation for this PR. Ideally our params system would support merging (and hopefully would), but that is IMO outside the scope of this PR. What I did is I detect the clash and report (hopefully) sane message and I believe it should suffice at this point. Would that work for you?